test/e2e/common/node/pod_level_resources.go

```diff
@@ -0,0 +1,332 @@
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		_, err := e2enode.GetRandomReadySchedulableNode(ctx, f.ClientSet)
+		framework.ExpectNoError(err)
+		if framework.NodeOSDistroIs("windows") {
+			e2eskipper.Skipf("not supported")
+		}
+	})
+	podLevelResourcesTests(f)
+})
+
+type containerInfo struct {
+	Name      string
+	Resources *resourceInfo
+}
+type resourceInfo struct {
+	CPUReq string
+	CPULim string
+	MemReq string
+	MemLim string
+}
+
+func makeContainer(info containerInfo) v1.Container {
+	cmd := []string{"/bin/sh", "-c", "sleep 1d"}
+	res := getResourceInfo(info.Resources)
+	return v1.Container{
+		Name:      info.Name,
+		Command:   cmd,
+		Resources: res,
+		Image:     imageutils.GetE2EImage(imageutils.BusyBox),
+		VolumeMounts: []v1.VolumeMount{
+			{
+				Name:      "sysfscgroup",
+				MountPath: "/tmp",
+			},
+		},
+	}
+}
+
+func getResourceInfo(info *resourceInfo) v1.ResourceRequirements {
+	var res v1.ResourceRequirements
+	if info != nil {
+		if info.CPUReq != "" || info.MemReq != "" {
+			res.Requests = make(v1.ResourceList)
+		}
+		if info.CPUReq != "" {
+			res.Requests[v1.ResourceCPU] = resource.MustParse(info.CPUReq)
+		}
+		if info.MemReq != "" {
+			res.Requests[v1.ResourceMemory] = resource.MustParse(info.MemReq)
+		}
+
+		if info.CPULim != "" || info.MemLim != "" {
+			res.Limits = make(v1.ResourceList)
+		}
+		if info.CPULim != "" {
+			res.Limits[v1.ResourceCPU] = resource.MustParse(info.CPULim)
+		}
+		if info.MemLim != "" {
+			res.Limits[v1.ResourceMemory] = resource.MustParse(info.MemLim)
+		}
+	}
+	return res
+}
+
+func makePod(metadata *metav1.ObjectMeta, podResources *resourceInfo, containers []containerInfo) *v1.Pod {
+	var testContainers []v1.Container
+	for _, container := range containers {
+		testContainers = append(testContainers, makeContainer(container))
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: *metadata,
+
+		Spec: v1.PodSpec{
+			Containers: testContainers,
+			Volumes: []v1.Volume{
+				{
+					Name: "sysfscgroup",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/cgroup"},
+					},
+				},
+			},
+		},
+	}
+
+	if podResources != nil {
+		res := getResourceInfo(podResources)
+		pod.Spec.Resources = &res
+	}
+
+	return pod
+}
+
+func VerifyPodResources(gotPod v1.Pod, expectedInfo resourceInfo) {
+	ginkgo.GinkgoHelper()
+	expectedResources := getResourceInfo(&expectedInfo)
+	gomega.Expect(gotPod.Spec.Resources).To(gomega.Equal(&expectedResources))
+}
+
+func VerifyQoS(gotPod v1.Pod, expectedQoS v1.PodQOSClass) {
+	ginkgo.GinkgoHelper()
+	gomega.Expect(gotPod.Status.QOSClass).To(gomega.Equal(expectedQoS))
+}
+
+func verifyPodCgroups(ctx context.Context, f *framework.Framework, pod *v1.Pod, expectedResources resourceInfo) error {
+	ginkgo.GinkgoHelper()
+	if pod.Spec.Resources == nil {
+		return nil
+	}
+	cmd := fmt.Sprintf("find /tmp -name '*%s*'", strings.ReplaceAll(string(pod.UID), "-", "_"))
```

Comment:
I found it really confusing seeing /tmp here. can you use a different path in the container, and also maybe pass around a variable for the location (so it's not hardcoded twice?)